src/runmode-dpdk.c

```diff
@@ -0,0 +1,1274 @@
+        errno = ERANGE;
+        SCReturnInt(-errno);
+    }
+
+    iconf->mempool_cache_size = entry_int;
+    SCReturnInt(0);
+}
+
+int ConfigSetRxDescriptors(DPDKIfaceConfig *iconf, intmax_t entry_int)
+{
+    if (entry_int <= 0) {
+        SCLogError(SC_ERR_DPDK_CONF, "Number of RX descriptors must be a positive number");
+        errno = ERANGE;
+        SCReturnInt(-errno);
+    }
+
+    iconf->nb_rx_desc = entry_int;
+    SCReturnInt(0);
+}
+
+int ConfigSetTxDescriptors(DPDKIfaceConfig *iconf, intmax_t entry_int)
+{
+    if (entry_int <= 0) {
+        SCLogError(SC_ERR_DPDK_CONF, "Number of TX descriptors must be a positive number");
+        errno = ERANGE;
+        SCReturnInt(-errno);
+    }
+
+    iconf->nb_rx_desc = entry_int;
+    SCReturnInt(0);
+}
+
+int ConfigSetMtu(DPDKIfaceConfig *iconf, intmax_t entry_int)
+{
+    if (entry_int < RTE_ETHER_MIN_MTU || entry_int > RTE_ETHER_MAX_JUMBO_FRAME_LEN) {
+        SCLogError(SC_ERR_DPDK_CONF, "Size of MTU must be between %" PRIu32 " and %" PRIu32,
+                RTE_ETHER_MIN_MTU, RTE_ETHER_MAX_JUMBO_FRAME_LEN);
+        errno = ERANGE;
+        SCReturnInt(-errno);
+    }
+
+    iconf->mtu = entry_int;
+    SCReturnInt(0);
+}
+
+int ConfigSetPromiscuousMode(DPDKIfaceConfig *iconf, int entry_bool)
+{
+    if (entry_bool)
+        iconf->flags |= DPDK_PROMISC;
+
+    SCReturnInt(0);
+}
+
+int ConfigSetMulticast(DPDKIfaceConfig *iconf, int entry_bool)
+{
+    if (entry_bool)
+        iconf->flags |= DPDK_MULTICAST; // enable
+
+    SCReturnInt(0);
+}
+
+int ConfigSetChecksumChecks(DPDKIfaceConfig *iconf, int entry_bool)
+{
+    if (entry_bool)
+        iconf->checksum_mode = CHECKSUM_VALIDATION_ENABLE;
+
+    SCReturnInt(0);
+}
+
+int ConfigSetChecksumOffload(DPDKIfaceConfig *iconf, int entry_bool)
+{
+    if (entry_bool)
+        iconf->flags |= DPDK_RX_CHECKSUM_OFFLOAD;
+
+    SCReturnInt(0);
+}
+
+int ConfigSetCopyIface(DPDKIfaceConfig *iconf, const char *entry_str)
+{
+    int retval;
+
+    if (entry_str == NULL || entry_str[0] == '\0' || strcmp(entry_str, "none") == 0) {
+        iconf->out_iface = NULL;
+        SCReturnInt(0);
+    }
+
+    retval = rte_eth_dev_get_port_by_name(entry_str, &iconf->out_port_id);
+    if (retval < 0) {
+        SCLogError(SC_ERR_DPDK_CONF, "Name of the copy interface (%s) not valid", entry_str);
+        errno = -retval;
+        SCReturnInt(retval);
+    }
+
+    iconf->out_iface = entry_str;
+    SCReturnInt(0);
+}
+
+int ConfigSetCopyMode(DPDKIfaceConfig *iconf, const char *entry_str)
+{
+    if (entry_str == NULL) {
+        SCLogError(SC_ERR_INVALID_VALUE, "Passed value is NULL");
+        errno = EINVAL;
+        SCReturnInt(-errno);
+    }
+
+    if (strcmp(entry_str, "none") == 0) {
+        iconf->copy_mode = DPDK_COPY_MODE_NONE;
+    } else if (strcmp(entry_str, "tap") == 0) {
+        iconf->copy_mode = DPDK_COPY_MODE_TAP;
+    } else if (strcmp(entry_str, "ips") == 0) {
+        iconf->copy_mode = DPDK_COPY_MODE_IPS;
+    } else {
```

Comment:
rework to include IDS mode

Reply:
you are mentioning IDS is default mode. that is good, but neither YAML or document for DPDK sections mentions these. Also enable `dpdk` does not call this out. rework